erts/emulator/beam/erl_process.c

```diff
@@ -7201,6 +7201,10 @@ resume_process(Process *p, ErtsProcLocks locks)
 					 &enq_prio,
 					 locks);
     add2runq(enqueue, enq_prio, p, state, NULL);
+
+    if (state & ERTS_PSFLG_PAUSED_TIMER) {
+        erts_resume_paused_proc_timer(p);
```

Comment:
This makes the lock-check unhappy, as this calls `erts_proc_sig_send_rpc_request()` which will try to acquire `proc_msgq` while holding `proc_status`. Acquiring both at the same time before calling `resume_process()` doesn't help as we'll then try to lock `proc_msgq` twice. I guess one option would be to extend `erts_proc_sig_send_rpc_request()` with an argument with the locks held, to check if it needs to acquire `proc_msgq` or not. Still, that will not be enough as there are calls to `erts_resume()` under mutexes that can't be acquired before `proc_msgq` (e.g. persistent-term updates). We could try to reorder the locks, but I guess it may be tricky to untangle.

Any recommendation on how to deal with this?

Reply:
Looking into this again; instead of using `erts_proc_sig_send_rpc_request()`, maybe we could use a system-task for this, that only requires holding `proc_status`, I think.